sentry/src/main/java/io/sentry/transport/RateLimiter.java

```diff
@@ -301,4 +328,39 @@ private long parseRetryAfterOrDefault(final @Nullable String retryAfterHeader) {
     }
     return retryAfterMillis;
   }
+
+  private void notifyRateLimitObservers(final boolean applied) {
+    for (IRateLimitObserver observer : rateLimitObservers) {
+      observer.onRateLimitChanged(applied);
+    }
+  }
+
+  public void addRateLimitObserver(@NotNull final IRateLimitObserver observer) {
+    rateLimitObservers.add(observer);
+  }
+
+  public void removeRateLimitObserver(@NotNull final IRateLimitObserver observer) {
+    rateLimitObservers.remove(observer);
+  }
+
+  @Override
+  public void close() throws IOException {
+    synchronized (timerLock) {
+      if (timer != null) {
+        timer.cancel();
+        timer = null;
+      }
+    }
```

Comment:
Probably makes sense to clear the list of observers, right?
```suggestion
        rateLimitObservers.clear();
    }
```